java/fury-core/src/main/java/org/apache/fury/serializer/ArraySerializers.java

```diff
@@ -208,6 +251,30 @@ private Object[] newArray(int numElements) {
       }
       return value;
     }
+
+    private GenericType buildCollectionGenericType(int dimensions) {
```

Comment:
I am hoping there is a better way to resolve this dyamically rather then using switch? Any suggestions?

Reply:
I think we can write using a loop, and using `TypeRef.of` to build generic type

Reply:
It's like:
```java

 Tuple2<Class<?>, Integer> info = TypeUtils.getArrayComponentInfo(declaredClass);
      List<TypeRef<?>> typeRefs = new ArrayList<>(info.f1 + 1);
      typeRefs.add(collectionTypeRef);
      for (int i = 0; i < info.f1; i++) {
        typeRefs.add(TypeUtils.getElementType(typeRefs.get(i)));
      }
      Collections.reverse(typeRefs);
      for (int i = 1; i < typeRefs.size(); i++) {
        TypeRef<?> arrayType = typeRefs.get(i - 1);
        TypeRef<?> typeRef =
            TypeRef.of(
                Array.newInstance(arrayType.getRawType(), 1).getClass(),
                typeRefs.get(i).getExtInfo());
        typeRefs.set(i, typeRef);
      }
```

See more details in `ClassDef#buildFieldType`

Reply:
Apologies if it is a dumb question, but where is collectionTypeRef coming from in this case? I was thinking it might be TypeUtils.COLLECTION_TYPE but even in this case its adding one element for looping over multiple dimensions so would still fail

Reply:
My bad, in your case, I think you can write like:
```java
TypeRef arrayTypeRef = TypeRef.of(componentClass);
for (int i = 0; i < dims -1; i++ ) {
  arrayTypeRef = collectionOf(arrayTypeRef);
}
```